### PR TITLE
feat: Add isEnabled property to SentrySDK (#697)

### PR DESCRIPTION
**Pull request by d2vydov**
## :scroll: Description

There is no a convenient way for check was the sentry started on client or not, 
it could be achieved in 4.4.x via Client.shared != nil

I've just added a property `Bool isEnabled` to SDK and Hub.

## :bulb: Motivation and Context

From comment of @bruno-garcia: 

> That's fair and would be a good addition.
> 
> On Android we have isEnabled:
> https://github.com/getsentry/sentry-android/blob/282d224739ee6536e7517821aab92f3587bed783/sentry-core/src/main/java/io/sentry/core/Sentry.java#L52

## :green_heart: How did you test it?

Unit tests/AppDelegate

## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify the changes
- [x] I've updated the CHANGELOG
- [x] No breaking changes

## :crystal_ball: Next steps

### CODE REVIEW

**Comment by d2vydov** [comment]
also, I can add a check `dsn` to `SentryClient` (`SentryFileManager` will fail with error if the `dsn` is invalid),  
and if it fails, i just init property with false. is it preferred? if yes, can I do this on separate PR?

**Comment by bruno-garcia** [comment]
What about we just go with flipping the flat (in SentrySDK) when `init` is called.
We can check if `DSN` is not available in the options or if it's a `""`, and not set to enabled in that case. 

Calling `init` creates a client and gets the SDK ready to go. Anything else is getting out of your way to disable it (like binding a client after the fact, that doesn't do anything, for example).

What do you think?

**Comment by d2vydov** [comment]
I agree, but I am not sure I understand what means "flipping the flat". 😀
So, anyway, I will change a code soon.

**Comment by bruno-garcia** [comment]
Sorry, I meant _flipping the flag_.
The flag here means the boolean field `_isEnabled`. Which would start with `@NO` and go to `@YES` as I described above.

**Comment by philipphofmann** [comment]
Thanks @d2vydov for opening up this PR and sorry for the late response. I think it is better to not have an extra field on the hub. I think the best way to do it right now is to check in the SDK if `nil != [[[[self currentHub] getClient] options] parsedDsn]`. We already check in the client [if the parsedDsn is nil](https://github.com/getsentry/sentry-cocoa/blob/7510ecfa834aa470332ed0e2c3c191431e65fe68/Sources/Sentry/SentryClient.m#L258-L261) and then just don't send any events. What do you think @bruno-garcia?

**Comment by bruno-garcia** [comment]
@philipphofmann that makes sense to me. It would be pretty accurate and simpler with that approach.

File: Sources/Sentry/include/SentryHub.h [hunk 1]
```diff
@@ -15,6 +15,7 @@ SENTRY_NO_INIT
 // Since there's no scope stack, single hub instance, experimenting with holding
 // session here.
 @property (nonatomic, readonly, strong) SentrySession *_Nullable session;
+@property (nonatomic, readonly, strong) NSNumber *isEnabled;
```
Comment (lhunath) [inline Sources/Sentry/include/SentryHub.h:18]: Any reason why this is an `NSNumber*` and not a `BOOL`; ie. type differs from `SentrySDK.isEnabled`?

File: Sources/Sentry/include/SentrySDK.h [hunk 2]
```diff
@@ -140,6 +140,11 @@ SENTRY_NO_INIT
  */
 @property (nonatomic, class, readonly) BOOL crashedLastRun;
 
+/**
+ * Checks if the SentrySDK was enabled or not.
+ */
+@property (nonatomic, class, readonly) BOOL isEnabled;
```
Comment (lhunath) [inline Sources/Sentry/include/SentrySDK.h:146]: It's important that there is a clean API for suspending and resuming the Sentry client, in order to smoothly conform with the privacy wishes that users have for the app & comply with GDPR and related privacy legislation.  Consequently, I believe there should be a setter for this property.

**Comment by philipphofmann** [comment]
We decided in https://github.com/getsentry/sentry-cocoa/issues/809 to undo #736, so this PR gets irrelevant. Thanks @d2vydov for opening it though.

**Pull request closed without merging**